python/pyspark/pandas/namespace.py

```diff
@@ -2814,9 +2824,18 @@ def to_numeric(arg):
     1.0
     """
     if isinstance(arg, Series):
-        return arg._with_new_scol(arg.spark.column.cast("float"))
+        if errors == "coerce":
+            return arg._with_new_scol(arg.spark.column.cast("int"))
```

Comment:
why should we change the type?

Reply:
I changed it to follow pandas' behavior below:

```python
>>> pd.to_numeric(pd.Series(['1', '2', '3']))
0    1
1    2
2    3
dtype: int64

>>> ps.to_numeric(ps.Series(['1', '2', '3']))
0    1.0
1    2.0
2    3.0
dtype: float32
```

But I'll revert this change since we cannot handle the case below properly with this change.

```python
>>> pd.to_numeric(pd.Series(['1.0', '2', '-3']))
0    1.0
1    2.0
2   -3.0
dtype: float64

>>> ps.to_numeric(ps.Series(['1.0', '2', '-3']))
0    1
1    2
2   -3
dtype: int32
```